code/__DEFINES/DNA.dm

```diff
@@ -100,8 +105,11 @@
 /// Can this species use the 'Body size' preference
 #define BODY_RESIZABLE 22
 
-#define MUTCOLORS2			23
-#define MUTCOLORS3			24
+#define MUTCOLORS2 22
+#define MUTCOLORS3 23
+
+#define HAIRCOLOR 24
+#define FACEHAIRCOLOR 25
```

Comment:
These are currently causing #741 on the testmerge. (To be fair, randomly testing Teshari feather colours isn't something that should be expected from a PR adding Vox)
One solution would be to add `HAIRCOLOR` and `FACEHAIRCOLOR` to the Teshari `species_traits` list, but a better way is to just remove the defines entirely.

This is something I looked in to when I was working on Teshari. Since both preferences are marked as `PREFERENCE_CATEGORY_SUPPLEMENTAL_FEATURES`, they won't ever be shown on the UI without another preference linking them, so restricting them to a trait isn't necessary.

Reply:
don’t want it removed for a certain reason, in it that we want species without actual hair to still have the hair color choice on prefs, and other external organs can use that color like vox hairs

Reply:
the tesh merge was completely not expected

Reply:
> don’t want it removed for a certain reason, in it that we want species without actual hair to still have the hair color choice on prefs, and other external organs can use that color like vox hairs

Removing `relevant_species_trait` does this, and it's actually how Teshari feather colouring works already.

If it isn't locked to a specific trait/feature then the preference will be available in *every* UI rather than just specific ones, and since it's 'supplemental' it will be added if any preferences are using it, and will stay unloaded in the background if not.

Reply:
having it appearing in species without hair/facehair and do basically nothing is not very useful

Reply:
It doesn't appear though. It's set to `PREFERENCE_CATEGORY_SUPPLEMENTAL_FEATURES` so the only times that it ever shows on the UI is when you open the tab for hair/feather/quill styles. If the species doesn't have any of those, then it'll just stay in the backend and never be visible.
`PREFERENCE_CATEGORY_SECONDARY_FEATURES`: (Permanently displayed on the right side of the UI)
![Secondary](https://user-images.githubusercontent.com/57483089/170894031-23a9655c-1916-4362-b669-4a389427ce11.png)
`PREFERENCE_CATEGORY_SUPPLEMENTAL_FEATURES`: (Only displayed inside the context menu of another separate preference)
![Supplemental](https://user-images.githubusercontent.com/57483089/170894044-4c588671-4533-4753-b361-d46d1e15dd3e.png)
<hr>

For supplemental features there's no frontend difference between restricting it to a species trait and not restricting it; both are exactly the same in that regard.
The advantage of adding a `relevant_species_trait` is that it won't be sent through `ui_data()` to the menu:
**Current:** *(From the TGUI debug console)*
![image](https://user-images.githubusercontent.com/57483089/170894931-de3d2372-4d13-4175-aa20-86bb032e5cd5.png)
**This PR:**
![image](https://user-images.githubusercontent.com/57483089/170894952-0888817b-3748-494f-be72-6d8a41d1333a.png)
(By itself that's a good thing since it makes the dev console a little bit cleaner.)

The *dis*advantage though, and why I suggested removing it, is that it's adding two full species trait defines in exchange for removing two lines of temporary UI data.


I don't want to get stuck on this or anything so if you want to keep it then that's fine. I just want to make sure that you're doing it in order to remove the unused UI data and *not* to restrict it to specific species, because that's the way it's working already.

Reply:
less stuff sent to ui_data is always better, the pref menu is already lagging